module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Logic/RecordsHelperTest.php

```diff
@@ -0,0 +1,339 @@
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFindTest\ILS\Logic;
+
+use PHPUnit\Framework\TestCase;
+use VuFind\Config\Config;
+use VuFind\ILS\Logic\RecordsHelper;
+use VuFind\Record\Loader;
+use VuFind\RecordDriver\AbstractBase;
+
+/**
+ * Unit tests for the RecordsHelper
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Sambhav Pokharel <sambhavpokharel@gmail.com>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class RecordsHelperTest extends TestCase
+{
+    /**
+     * Test getDrivers with empty array
+     *
+     * @return void
+     */
+    public function testGetDriversWithEmptyArray(): void
+    {
+        $config = $this->createMock(Config::class);
+        $loader = $this->createMock(Loader::class);
+
+        $loader->expects($this->never())
+            ->method('loadBatch');
+
+        $helper = new RecordsHelper($config, $loader);
+        $result = $helper->getDrivers([]);
+
+        $this->assertIsArray($result);
+        $this->assertEmpty($result);
+    }
+
+    /**
+     * Test getDrivers with valid records
+     *
+     * @return void
+     */
+    public function testGetDriversWithValidRecords(): void
+    {
+        $config = $this->createMock(Config::class);
+        $loader = $this->createMock(Loader::class);
+
+        $records = [
+            ['id' => 'record1', 'source' => 'Solr', 'title' => 'Test Book 1'],
+            ['id' => 'record2', 'source' => 'Solr', 'title' => 'Test Book 2'],
+            ['id' => 'record3', 'source' => 'VuFind', 'title' => 'Test Book 3'],
+        ];
+
+        $expectedIds = [
+            ['id' => 'record1', 'source' => 'Solr'],
+            ['id' => 'record2', 'source' => 'Solr'],
+            ['id' => 'record3', 'source' => 'VuFind'],
+        ];
+
+        $driver1 = $this->createMock(AbstractBase::class);
+        $driver2 = $this->createMock(AbstractBase::class);
+        $driver3 = $this->createMock(AbstractBase::class);
+
+        $mockDrivers = [$driver1, $driver2, $driver3];
+
+        $loader->expects($this->once())
+            ->method('loadBatch')
+            ->with($expectedIds, true)
+            ->willReturn($mockDrivers);
+
+        $driver1->expects($this->once())
+            ->method('setExtraDetail')
+            ->with('ils_details', $records[0]);
+
+        $driver2->expects($this->once())
+            ->method('setExtraDetail')
+            ->with('ils_details', $records[1]);
+
+        $driver3->expects($this->once())
+            ->method('setExtraDetail')
+            ->with('ils_details', $records[2]);
+
+        $helper = new RecordsHelper($config, $loader);
+        $result = $helper->getDrivers($records);
+
+        $this->assertIsArray($result);
+        $this->assertCount(3, $result);
+        $this->assertSame($mockDrivers, $result);
```

Comment:
The assertSame should implicitly cover the isArray and Count, so I don't think we need to make those extra assertions -- if it's the same as the expected array, we already know the type and the count.

```suggestion
        $this->assertSame($mockDrivers, $result);
```